lib/model/doctrine/ParlementaireScrutin.class.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+class ParlementaireScrutin extends BaseParlementaireScrutin
+{
+
+  public function setParlementaire($id_an) {
+    $parl = Doctrine::getTable('Parlementaire')->findOneByIdAn($id_an);
+    if (!$parl) {
+      throw new Exception("Aucun parlementaire trouvé avec l'ID AN $id_an");
+    }
+
+    return $this->_set('parlementaire_id', $parl->id);
+  }
+
+  public function setScrutin($scrutin) {
+    return $this->_set('scrutin_id', $scrutin->id);
+  }
+
+  public function updatePresence() {
+    $this->Scrutin->Seance->setUnsetPresenceLight(
+      $this->parlementaire_id,
+      $this->parlementaire_groupe_acronyme,
+      'scrutin',
+      $this->Scrutin->getLinkSource(),
+      $this->position != 'nonVotant' && $this->mise_au_point_position == NULL
```

Comment:
je pense que s'il y a bien une position passée sans délégation et une mise au point ça correspond bien à une présence, en revanche j'ai l'impression que par_delegation n'est jamais utilisé jusqu'ici pour confirmer les présences, je suppose donc que c'est une confusion ici et qu'il faut remplacer mise_au_point_position par par_delegation

Reply:
Par ailleurs non votant indique j'ai l'impression le président, il est donc a priori bien présent, je pense que le bon check est position not Null et par_delegation False